redis.go

```diff
@@ -677,16 +677,16 @@ func NewClient(opt *Options) *Client {
 func (c *Client) init() {
 	c.cmdable = c.Process
 	c.initHooks(hooks{
-		dial:       c.baseClient.dial,
-		process:    c.baseClient.process,
-		pipeline:   c.baseClient.processPipeline,
-		txPipeline: c.baseClient.processTxPipeline,
+		dial:       c.dial,
+		process:    c.process,
+		pipeline:   c.processPipeline,
+		txPipeline: c.processTxPipeline,
```

Comment:
I think having the embedded field here (`baseClient`) makes it easier to follow the logic and it is more clear what is happening in the client. Let's alter the linters in such a way, to allow us to have the embedded fields in the selectors. I did try different settings for the `staticcheck` linter and found the following to my liking:
```
    staticcheck:
      checks:
        - all
        # Incorrect or missing package comment.
        # https://staticcheck.dev/docs/checks/#ST1000
        - -ST1000
        # Omit embedded fields from selector expression.
        # https://staticcheck.dev/docs/checks/#QF1008
        - -QF1008
```

You can try by adding those in a section `settings:` in the `linters:` section of the `.golangci.yml`. Let me know what you think.